db/view/view.cc

```diff
@@ -189,9 +188,15 @@ db::view::base_info_ptr view_info::make_base_dependent_view_info(const schema& b
     std::vector<column_id> base_regular_columns_in_view_pk;
     std::vector<column_id> base_static_columns_in_view_pk;
 
-    _has_same_partition_key = boost::equal(
-            base.partition_key_columns() | boost::adaptors::transformed(std::mem_fn(&column_definition::name)),
-            _schema.partition_key_columns() | boost::adaptors::transformed(std::mem_fn(&column_definition::name)));
+    _is_partition_key_permutation_of_base_partition_key =
+        boost::algorithm::all_of(base.partition_key_columns(), [this] (const column_definition& base_col) {
+            const column_definition *view_col = _schema.get_column_definition(base_col.name());
```

Comment:
Nit: our coding style suggests putting `*`, `&` and `&&` near the type name and not the variable. So:

```c++
const column_definition* view_col = /* ... */
```